File: dagger_app/src/main/java/my/liewjuntung/daggerviperusage/dagger/AppModule.java
package my.liewjuntung.daggerviperusage.dagger;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by pandawarrior on 24/10/2017.
 */
@Module
public class AppModule {

    /**
     * Provides application level context
     * @param application
     * @return
     */
    @Provides
    @Singleton
    public Context provideContext(Application application) {
        return application;
    }
}

File: kotlin_dagger_app/src/main/java/my/liewjuntung/kotlindaggerapp/dagger/AppModule.kt
package my.liewjuntung.kotlindaggerapp.dagger

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by pandawarrior on 24/10/2017.
 */
@Module
class AppModule {

    /**
     * Provides application level context
     * @param application
     * @return
     */
    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }
}
